lib/ansible/modules/system/mount.py

```diff
@@ -666,6 +702,25 @@ def main():
                         msg="Error unmounting %s: %s" % (name, msg))
 
             changed = True
+    elif state == 'mount':
+        if not os.path.exists(name) and not module.check_mode:
+            try:
+                os.makedirs(name)
+            except (OSError, IOError):
+                e = get_exception()
+                module.fail_json(
+                    msg="Error making dir %s: %s" % (name, str(e)))
+
+        res = 0
+
+        # TODO: add a check to see if src have changed and remount if necessary
```

Comment:
This would be nice to have implemented from the beginning.

Reply:
I'm currently working on this.

Reply:
Remounting would require another option to do / not do a remount.

Reply:
Why do you think so?

Reply:
For the moment I have done so that when the device is already mounted, it does not touch it. Do you find it acceptable this way?

I tried to add detection of changed settings, but i ended up thinking it would require to modify others functions (may be add new ones) and i'm not comfortable with this…